### DISCUSSION TITLE
Difference between Transformer and Mutator

### DISCUSSION THREAD

**Discussion by Hypenate** [orval-labs/orval#427] [Q&A] [answered]
Our backend is producing _snake case_ but I want to transform (or mutate?) all the classes and properties to _camel case_.
I want to remove the **_** and make some characters upercase etc.

But where do I do this? And how can I change the propertie names?
I also saw that there is a transformer under output but also under input?

I'm a bit confused atm 🤔

**Comment by Hypenate** [comment, marked as answer]
It seems the mutator is able to change the actualy request/response to the API
Where as the transformer is able to transform the generation of the yaml file.

**Comment by anymaniax** [comment]
@Hypenate Hi, You are right the transform are at generation time and mutators are used at run time
